gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/AuthenticationFlowHandlerTest.java

```diff
@@ -198,18 +221,20 @@ public void shouldRedirectToMFAEnrollmentPage_adaptiveMFA_no_enroll() throws Exc
     @Test
     public void shouldNoRedirectToMFAChallengePage_adaptiveMFA_no_active_enroll_and_endUser_not_enrolled() throws Exception {
         router.route().order(-1).handler(rc -> {
+            String rule = "{context.attributes['geoip']['country_iso_code'] == 'FR'";
+            EnrollSettings enrollSettings = createEnrollSettings(true, false, MfaEnrollType.REQUIRED, 0, "");
+            ChallengeSettings challengeSettings = createChallengeSettings(true, MfaChallengeType.REQUIRED, rule);
```

Comment:
**todo**: the test is named 
 shouldNoRedirectToMFAChallengePage_**adaptiveMFA**_no_active_enroll_and_endUser_not_enrolled but the ChallengeType is set to REQUIRED, should be CONDITIONAL